### PR TITLE
Added filestatus sink

### PR DESCRIPTION
**Pull request by acnsalb**

### CODE REVIEW

**Comment by acnsalb** [comment]
@MarkusAmshove please approve workflow (while I'm trying to get Sonarqube working locally)

**Comment by acnsalb** [comment]
@MarkusAmshove with the number of changed lines in CliAnalyzer.java, I will never reach 80% without testing that class. Any ideas?

**Comment by MarkusAmshove** [comment]
> @MarkusAmshove with the number of changed lines in CliAnalyzer.java, I will never reach 80% without testing that class. Any ideas? 

If you say the PR is ready to be merged I can do a pass of manual testing again and merge it regardless of the coverage :)

**Comment by acnsalb** [comment]
> > @MarkusAmshove with the number of changed lines in CliAnalyzer.java, I will never reach 80% without testing that class. Any ideas?
> 
> If you say the PR is ready to be merged I can do a pass of manual testing again and merge it regardless of the coverage :)

It is. I changed the PR status from draft to open.

File: libs/natlint/src/main/java/org/amshove/natlint/linter/LinterContext.java [hunk 1]
```diff
@@ -75,30 +75,23 @@ public void updateEditorConfig(EditorConfig config)
 
 	void analyze(ISyntaxNode syntaxNode, IAnalyzeContext context)
 	{
-		try
```
Comment (MarkusAmshove) [inline libs/natlint/src/main/java/org/amshove/natlint/linter/LinterContext.java:78]: I think these changes are from an older `main` branch, can you update this file to the current `main`?
[The current version](https://github.com/MarkusAmshove/natls/blob/main/libs/natlint/src/main/java/org/amshove/natlint/linter/LinterContext.java#L76-L102) has this safeguard specifically mentioned [in a commit](https://github.com/MarkusAmshove/natls/commit/c385571d7428f4167c04574b8131072dc8453a74)

**Review by MarkusAmshove — CHANGES_REQUESTED** [review]
Changes look good, I'll merge this when the changes to `LinterContext` are  reverted :+1:

File: libs/natlint/src/main/java/org/amshove/natlint/linter/LinterContext.java (hunk 1 above)
Comment (acnsalb) [reply libs/natlint/src/main/java/org/amshove/natlint/linter/LinterContext.java:78]: This is something I changed. I now catch the error and put it in the filestatuses.csv.
So it's handled from the cli POV, but I guess I'm missing handling from the API POV.

Comment (MarkusAmshove) [reply libs/natlint/src/main/java/org/amshove/natlint/linter/LinterContext.java:78]: Looking at the current implementation in main, it doesn't continue to analyze when an exception occurs. We can ignore my comment and do graceful handling in the LSP :)

**Review by MarkusAmshove — APPROVED** [review]

**Pull request merged**